cpp/src/neighbors/batch_ann/batch_ann.cu

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "batch_ann.cuh"
+#include "cuvs/neighbors/common.hpp"
+
+namespace cuvs::neighbors::batch_ann {
```

Comment:
Since you've redefined a new interface, it means these all need to be explicitly compiled. If you instead just have the single-gpu apis dispatch to the batched code when the settings meet the proper criteria, you won't need to explicitly instantiate the APIs.

Reply:
I'm guessing this comment is from _before_ we decided to expose this as a public API?